Sources/ProjectDescription/TargetDependency.swift

```diff
@@ -2,11 +2,16 @@ import Foundation
 
 // MARK: - TargetDependency
 
+public enum SDKStatus: String {
+    case required, optional
+}
+
 public enum TargetDependency: Codable {
     case target(name: String)
     case project(target: String, path: String)
     case framework(path: String)
     case library(path: String, publicHeaders: String, swiftModuleMap: String?)
+    case sdk(name: String, status: SDKStatus)
 
```

Comment:
What do you think about providing a default status? The majority of use cases I've come across had a `required` status (keen to hear your thoughts on this, is it a valid assumption?)

Default values for enum associated types is coming ([SE-0155](https://github.com/apple/swift-evolution/blob/master/proposals/0155-normalize-enum-case-representation.md)) - in the interim we could write a static initialiser to achieve the same effect:

```swift
public static sdk(name: String) -> TargetDependency {
    return .sdk(name: name, status: .required)
}
```

Reply:
Sure I'll add this.

Reply:
Agree here :)